recipe_scrapers/_exceptions.py

```diff
@@ -41,3 +41,17 @@ class SchemaOrgException(RecipeScrapersExceptions):
 
     def __init__(self, message):
         super().__init__(message)
+
+
+class DisallowedByRobotsTXT(RecipeScrapersExceptions):
+    """Retrieval of the URL is disallowed by the host's robots.txt file"""
+
+    def __init__(self, url):
+        self.url = url
+        message = str().join(
```

Comment:
nit: Is there a reason to join like this instead of writing a multi-line string or some other simpler implementation?

Reply:
Oops, I forgot to respond to this.  Not really, no: I think I sometimes do this to avoid ambiguity around spacing from start-of-line to start-of-text within a string.